lib/api/core/plugins/pluginsManager.js

```diff
@@ -175,33 +183,30 @@ class PluginsManager {
           .timeout(initTimeout)
           .then(initStatus => {
             if (initStatus === false) {
-              return Bluebird.reject(new Error(`Something went wrong during the initialization of the plugin ${pluginName}.`));
+              return reject(new PluginImplementationError(`Something went wrong during the initialization of the plugin "${pluginName}".`));
             }
-            if (plugin.object.hooks) {
-              initHooks(plugin, this.kuzzle);
+
+            if (plugin.object.controllers && !initControllers(this.controllers, this.routes, plugin, pluginName)) {
+              return reject(new PluginImplementationError(`Unable to load plugin "${pluginName}": controllers are malformed`));
             }
 
-            if (plugin.object.pipes) {
-              initPipes(this, plugin, pipeWarnTime, pipeTimeout);
+            if (plugin.object.strategies && !injectAuthentication(this.kuzzle, this.authentications, plugin.object, pluginName)) {
+              return reject(new PluginImplementationError(`Unable to load plugin "${pluginName}": authentications strategies are malformed`));
             }
 
-            if (plugin.object.controllers) {
-              initControllers(this.controllers, this.routes, plugin, pluginName);
+            if (plugin.object.hooks) {
+              initHooks(plugin, this.kuzzle);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the nested one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ACollapsibleIfStatements)